File: Java/program/Clothing.java
// Kelas Anak
public class Clothing extends Product{
    // Private attributes
    private String size;
    private String material;
    private String gender;

    // Konstruktor untuk nilai default pada attribut
    public Clothing(){
        this.size = "";
        this.material = "";
        this.gender = "";
    }

    // Konstruktor untuk nilai yang disesuaikan parameter pada attribut
    public Clothing(String size, String material, String gender){
        this.size = size;
        this.material = material;
        this.gender = gender;
    }

    // Get size
    public String get_size(){
        return this.size;
    }

    // Get material
    public String get_material(){
        return this.material;
    }

    // Get gender
    public String get_gender(){
        return this.gender;
    }

    // Set size
    public void set_size(String size){
        this.size = size;
    }

    // Set material
    public void set_material(String material){
        this.material = material;
    }

    // Set gender
    public void set_gender(String gender){
        this.gender = gender;
    }
}

File: CPP/program/Clothing.cpp
// Library dari induk
#include "Product.cpp"

// Class Clothing, anak dari class Product
class Clothing : public Product
{
    // Atribut private yang dimiliki class Clothing
    private:
        string size, material, gender;

    // Method yang bisa diakses diluar
    public:
        Clothing() {
            this->size = "";
            this->material = "";
            this->gender = "";
        }
        Clothing(string size, string material, string gender) {
            this->size = size;
            this->material = material;
            this->gender = gender;
        }

        // Get size 
        string get_size() {
            return this->size;
        }

        // Set size
        void set_size(string size) {
            this->size = size;
        }

        // Get material 
        string get_material() {
            return this->material;
        }

        // Set material
        void set_material(string material) {
            this->material = material;
        }

        // Get gender 
        string get_gender() {
            return this->gender;
        }

        // Set gender
        void set_gender(string gender) {
            this->gender = gender;
        }

        ~Clothing(){

        }
};